### DISCUSSION TITLE
Save detection results as a text file

### DISCUSSION THREAD

**Discussion by askppp** [open-mmlab/mmdetection#9256] [Q&A]
Hello,

i'm running an interference on an image. 

I save the result of the detections with "model.show_result" and i get the result image with drawn bboxes and confidence scores.

Is there a script to get the result of the detections on the image saved in a **text file** like this :

**Bboxes Coordinates  / Confidence Score / Class**

Best regards.

**Comment by Maenansyn** [comment]
inference_detector() returns the [results ](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/apis/inference.py#L162) before those are shown by show_result_pyplot(). You could save it in-between or after the pyplot drawing call. The result is normally structured like this array:
Dimension 1: List of imgs
Dimension 2: List of categories in img
Dimension 3 (for each category): List of predicted objects {category_id, bbox[x1,y1,x2y2], confidence_score}

So similarly as you want it, just with separation by category added. You can see the structure [here](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/coco.py#L239).
